src/initialize.F90

```diff
@@ -722,76 +723,15 @@ subroutine adjust_indices()
       ! =======================================================================
       ! ADJUST INDICES FOR EACH TALLY FILTER
 
-      FILTER_LOOP: do j = 1, t%n_filters
-
-        select case (t%filters(j)%type)
-        case (FILTER_DISTRIBCELL)
-          do k = 1, size(t%filters(j)%int_bins)
-            id = t%filters(j)%int_bins(k)
-            if (cell_dict%has_key(id)) then
-              t%filters(j)%int_bins(k) = cell_dict%get_key(id)
-            else
-              call fatal_error("Could not find cell " // trim(to_str(id)) // &
-                               " specified on tally " // trim(to_str(t%id)))
-            end if
-
-          end do
-        case (FILTER_CELL, FILTER_CELLBORN)
-
-          do k = 1, t%filters(j)%n_bins
-            id = t%filters(j)%int_bins(k)
-            if (cell_dict%has_key(id)) then
-              t%filters(j)%int_bins(k) = cell_dict%get_key(id)
-            else
-              call fatal_error("Could not find cell " // trim(to_str(id)) &
-                   &// " specified on tally " // trim(to_str(t%id)))
-            end if
-          end do
-
-        case (FILTER_SURFACE)
+      FILTER_LOOP: do j = 1, t % n_filters
```

Comment:
I'd suggest replacing most uses of `t % n_filters` with `size(t % filters)`. I'd like to get away from these n_ variables where they just indicate the size of an allocated array.